blockchain/types/transaction_test.go

```diff
@@ -193,17 +310,48 @@ func TestEIP2718TransactionEncode(t *testing.T) {
 	}
 }
 
-func TestTransactionEncode(t *testing.T) {
-	txb, err := rlp.EncodeToBytes(rightvrsTx)
-	if err != nil {
-		t.Fatalf("encode error: %v", err)
-	}
-	should := common.FromHex("f86103018207d094b94f5374fce5edbc8e2a8697c15331677e6ebf0b0a82554426a098ff921201554726367d2be8c804a7ff89ccf285ebc57dff8ae4c44b9c19ac4aa08887321be575c8095f789dd4c743dfe42c1820f9231f98a962b210e3ac2452a3")
-	if !bytes.Equal(txb, should) {
-		t.Errorf("encoded RLP mismatch, ɡot %x", txb)
+func TestEIP1559TransactionEncode(t *testing.T) {
+	// RLP representation
+	{
+		have, err := rlp.EncodeToBytes(signedEip1559Tx)
+		if err != nil {
+			t.Fatalf("encode error: %v", err)
+		}
+		want := common.FromHex("02f89d010301018261a894b94f5374fce5edbc8e2a8697c15331677e6ebf0b0a825544f838f7940000000000000000000000000000000000000001e1a0000000000000000000000000000000000000000000000000000000000000000001a0c9519f4f2b30335884581971573fadf60c6204f59a911df35ee8a540456b2660a032f1e8e2c5dd761f9e4f88f41c8310aeaba26a8bfcdacfedfa12ec3862d37521")
+		if !bytes.Equal(have, want) {
+			t.Errorf("encoded RLP mismatch, got %x", have)
+		}
 	}
 }
 
+func TestEffectiveGasPrice(t *testing.T) {
+	baseFee := big.NewInt(2000)
```

Comment:
How about adding test case with 0 baseFee?

Reply:
I added this case which the baseFee has 0. Would you check it?